### ISSUE TITLE
"case.submit" failed to restart a multiple instance simulation 

### ISSUE THREAD

**Issue by minxu74** [ESMCI/cime#4700]
Tried to restart a multiple instance simulation and 'case.submit' failed with the following the error:

```
  File "/global/u2/m/minxu/MyGit/e3sm_v3/cime/CIME/case/case_submit.py", line 283, in submit
    run_and_log_case_status(
  File "/global/u2/m/minxu/MyGit/e3sm_v3/cime/CIME/status.py", line 88, in run_and_log_case_status
    rv = func()
         ^^^^^^
  File "/global/u2/m/minxu/MyGit/e3sm_v3/cime/CIME/case/case_submit.py", line 267, in <lambda>
    functor = lambda: _submit(
                      ^^^^^^^^
  File "/global/u2/m/minxu/MyGit/e3sm_v3/cime/CIME/case/case_submit.py", line 73, in _submit
    if case.get_value("MULTI_DRIVER") and case.get_value("NINST") > 1:
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

The error was caused by case.get_value("NINST") as the NINST in env_mach_pes.xml does not have any value. In order to get the NINST value, a "subgroup" is needed, like `case.get_value("NINST", subgroup="ATM")`. But I do not know the logic here. Why need to get the NINST value? Why only check one file **rpointer.drv_0001** in the `case_submit.py` code.

**Comment by jedwards4b** [comment]
@minxu74  Thank you for reporting, I'll have a fix shortly.

**Comment by minxu74** [comment]
@jedwards4b Thanks. If the intention is to confirm that restart files are in the run directory,  do we need to check all rpointer files of instances? instead of just checking the first instance drv rpointer file?

**Comment by jedwards4b** [comment]
If you want to write and submit that code you are welcome to do so.  This is just intended to be a sanity check.

**Comment by minxu74** [comment]
Thanks. Based on the code:
```
        # Variable MULTI_DRIVER is always true for nuopc so we need to also check NINST > 1
        if case.get_value("MULTI_DRIVER") and case.get_value("NINST") > 1:
            rpointer = rpointer + "_0001"
        expect(
            os.path.exists(os.path.join(rundir, rpointer)),
            "CONTINUE_RUN is true but this case does not appear to have restart files staged in {} {}".format(
                rundir, rpointer
            ),
        )
```
It only makes sure the `rpointer.drv_0001` file in the run directory and does not the check if the restart file in it is in the run directory.  It shall be easy to loop over the number of instance to check existences of all rpointer.drv files.  For rpointer files of other components, they depends on activation of components. I do not know how to do it.  So I can submit codes to check rpointer.drv files from all instances.

**Comment by jedwards4b** [comment]
I would suggest that you start from #4701 that addresses the original issue.

**Comment by gold2718** [comment]
Have you tried replacing `case.get_value("NINST")` with `int(case.get_value("NINST"))`?
That worked for me (but using a different version of CIME).

**Comment by jedwards4b** [comment]
@gold2718 You may have noticed that I fixed this in #4701,  @minxu74 is correct that the mct driver has no variable NINST - it is only defined for nuopc.
